src/main/java/org/broadinstitute/hellbender/tools/exome/copyratio/CopyRatioData.java

```diff
@@ -0,0 +1,112 @@
+package org.broadinstitute.hellbender.tools.exome.copyratio;
+
+import com.google.common.primitives.Doubles;
+import org.apache.commons.math3.stat.descriptive.moment.Mean;
+import org.apache.commons.math3.stat.descriptive.moment.Variance;
+import org.broadinstitute.hellbender.tools.exome.ReadCountRecord;
+import org.broadinstitute.hellbender.tools.exome.SegmentedGenome;
+import org.broadinstitute.hellbender.tools.exome.TargetCollection;
+import org.broadinstitute.hellbender.utils.mcmc.DataCollection;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * {@link DataCollection} for the copy-ratio model containing the set of coverages and the grouping of coverages into segments.
+ *
+ * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
+ */
+public final class CopyRatioData implements DataCollection {
+    private final int numSegments;
+    private final int numTargets;
+    private final double coverageMin;
+    private final double coverageMax;
+    private final double coverageRange;
+
+    private final List<List<Double>> coveragesPerSegment = new ArrayList<>();
+    private final List<Integer> numTargetsPerSegment;
+    private final List<Integer> startTargetsPerSegment = new ArrayList<>();
+
+    public CopyRatioData(final SegmentedGenome segmentedGenome) {
+        if (segmentedGenome.getGenome().getTargets().targetCount() == 0) {
```

Comment:
Consider using the GATK public method `Utils.validateArg(segmentedGenome.getGenome().getTargets().targetCount() > 0, "Cannot construct CopyRatioData with no target-coverage data.")`